### PR TITLE
fix: update template item to show loading indicator on selected template only

### LINKED ISSUE #29135: Bug - All templates show loader when `use this template` is clicked.
* when we click on any "building block" or "template", all the buttons on different cards start showing the loading state. 
* [Video link](https://www.loom.com/share/af217de7ddda48a2ac37e9b9a8d60162)

We should just show loader state on the one that we have forked and disable it for rest of them

### PR DESCRIPTION
**Pull request by jacquesikot**
## Description
This PR addresses the issue where the loading indicator is erroneously triggered for all template cards when a user attempts to fork a single template. 

The proposed changes in this PR ensure that the loading indicator is appropriately tied to the specific template selected by the user. By isolating the indicator to the clicked template, we enhance the user experience and eliminate any confusion related to the forking process.

#### PR fixes following issue(s)
Fixes #29135 

#### Media

- Bug fix (non-breaking change which fixes an issue)

## Testing

- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Implemented a new state management feature to track active template operations, enhancing the user experience during template import and forking processes.
  - Introduced a new button state to prevent conflicting operations, ensuring a smoother template handling experience.

- **Enhancements**
  - Improved the visual feedback on the template interface with updated button states to reflect ongoing operations.

- **Bug Fixes**
  - Resolved an issue where users could initiate multiple conflicting template operations simultaneously.

### CODE REVIEW

**Review by rahulbarwal — APPROVED** [review]
Code changes look fine, lets test this out on DP before merging.

**Review by rahulbarwal — CHANGES_REQUESTED** [review]
Tested this on DP. 

We are not showing loading symbol on other templates now, but they still seem enabled. 
Can we have this behavior:
1. Show loading symbol + disable on the one we clicked on.
2. Disable on the others

**Review by rahulbarwal — APPROVED** [review]

**Comment by rahulbarwal** [comment]
@jacquesikot Not able to see disabled buttons in the DP: https://ce-29252.dp.appsmith.com/
can you please check one more time?

File: app/client/src/pages/Templates/Template/index.tsx [hunk 1]
```diff
@@ -113,6 +114,8 @@ export function TemplateLayout(props: TemplateLayoutProps) {
 
   const onForkButtonTrigger = (e: React.MouseEvent<HTMLElement>) => {
     if (props.onForkTemplateClick) {
+      if (isImportingTemplateToApp) return;
+      setSelectedTemplateId(id);
```
Comment (AmanAgarwal041) [inline app/client/src/pages/Templates/Template/index.tsx:118]: Where are we resetting it in case the forking of the template fails for some reason ?

Comment (jacquesikot) [reply app/client/src/pages/Templates/Template/index.tsx]: Thanks Aman, I have moved the state to the templateReducer and reset it on IMPORT_TEMPLATE_TO_APPLICATION_ONBOARDING_FLOW_SUCCESS and IMPORT_TEMPLATE_TO_APPLICATION_ONBOARDING_FLOW_ERROR

**Review by rahulbarwal — APPROVED** [review]

**Pull request merged**